src/nvim/os/fs.c

```diff
@@ -563,8 +568,13 @@ ptrdiff_t os_write(const int fd, const char *const buf, const size_t size)
   size_t written_bytes = 0;
   while (written_bytes != size) {
     assert(size >= written_bytes);
+#ifdef WIN32
+    const ptrdiff_t cur_written_bytes =
+      write(fd, buf + written_bytes, (unsigned int)(size - written_bytes));
```

Comment:
would rather live with the warning than have this kind of code duplication. Same for the one above. Until a better solution is found ..